app.py

```diff
@@ -11,700 +11,304 @@
-            None,
-            "avail",
-        )
-    
-    elif submission_indicator_val == "Done":
-        return built_sched, e_card, "sched"
-    
-    elif submission_indicator_val == "Submitting... please wait":
+    run_click: int,
+    shifts_per_employee: list[int],
+    employees_per_shift: list[int],
+    checklist: list[int],
+    consecutive_shifts: int,
+    sched_df: DataFrame,
+) -> tuple[DataFrame, bool]:
+    """Run a job on the hybris solver when the run button is clicked."""
+    if run_click == 0 or ctx.triggered_id != "run-button":
+        raise PreventUpdate
+
+    if ctx.triggered_id == "run-button":
         shifts = list(sched_df["props"]["data"][0].keys())
         shifts.remove("Employee")
-        availability = utils.availability_to_dict(
-            sched_df["props"]["data"], shifts
-        )
+        availability = utils.availability_to_dict(sched_df["props"]["data"], shifts)
         employees = list(availability.keys())
 
-        if 1 in checklist_value:
-            isolated = True
-        else:
-            isolated = False
-
-        if 2 in checklist_value:
-            manager = True
-        else:
-            manager = False
-
-   
-        
-        print("\nBuilding CQM...\n")
-        min_shifts = s_per_e_range[0]
-        max_shifts = s_per_e_range[1]
-        [shift_min, shift_max] = e_per_s_range
+        isolated_days_allowed = True if 0 in checklist else False
+        manager_required = True if 1 in checklist else False
+
         cqm = employee_scheduling.build_cqm(
             availability,
             shifts,
-            min_shifts,
-            max_shifts,
-            shift_min,
-            shift_max,
-            manager,
-            isolated,
-            k + 1,
+            *shifts_per_employee,
+            *employees_per_shift,
+            manager_required,
+            isolated_days_allowed,
+            consecutive_shifts + 1,
         )
 
-        print("\nSubmitting CQM...\n")
         feasible_sampleset, errors = employee_scheduling.run_cqm(cqm)
-
         sample = feasible_sampleset.first.sample
 
         sched = utils.build_schedule_from_sample(sample, shifts, employees)
 
-        if errors is None:
-            new_card_body = " "
-        else:
-            new_card_body = [
-                html.H4(
-                    children="Issues with Schedule", style={"font-family": ff}
-                ),
-                html.Div(errors, style={"whiteSpace": "pre-wrap"}),
-            ]
-
-        now = datetime.now()
-        month = now.month
-        year = now.year
-
         return (
-            utils.display_schedule(sched, availability, month, year),
-            new_card_body,
-            "sched",
+            utils.display_schedule(sched, availability, NOW.month, NOW.year),
+            False,
         )
 
-    else:
-        return no_update, no_update, no_update
+    return no_update
```

Comment:
Oh that's cool, I didn't know you could just return one no_update for the whole thing

Reply:
Oh, I didn't know that either (and I'm not sure this is actually ever called). I probably just forgot to change to `raise PreventUpdate` or `return no_update, no_update`. We could probably just remove this along with the `if ctx.triggered_id == "run-button"` altogether since the only input is the run-button.